From 0697d8f69d760413732c5d9cd18e38c75d7fdec5 Mon Sep 17 00:00:00 2001
From: rui guo <rui.guo@amlogic.com>
Date: Thu, 16 Feb 2023 18:47:11 +0800
Subject: [PATCH 7/7] xtest: pkcs11_1014: fix CKA_VALUE_LEN type

Fixes type used to store a CKA_VALUE_LEN attribute value, that is
CK_ULONG as per PKCS#11 specification, no uint32_t.

The issue was found on 64-bit xtest where
pkcs11_1000.c:3496: Expression "g_decrypt == CK_TRUE" (0 == 1) is false
pkcs11_1014.1 FAILED

Signed-off-by: rui guo <rui.guo@amlogic.com>
Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>
Reviewed-by: Etienne Carriere <etienne.carriere@linaro.org>
---
 host/xtest/pkcs11_1000.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/host/xtest/pkcs11_1000.c b/host/xtest/pkcs11_1000.c
index 4b36909..9c825af 100644
--- a/host/xtest/pkcs11_1000.c
+++ b/host/xtest/pkcs11_1000.c
@@ -3403,7 +3403,7 @@ static void xtest_pkcs11_test_1014(ADBG_Case_t *c)
 	CK_BBOOL g_decrypt = CK_FALSE;
 	CK_BBOOL g_wrap = CK_FALSE;
 	CK_BBOOL g_unwrap = CK_FALSE;
-	uint32_t g_len = 0;
+	CK_ULONG g_len = 0;
 	CK_ATTRIBUTE get_template[] = {
 		{ CKA_LABEL, (CK_UTF8CHAR_PTR)g_label, sizeof(g_label) },
 		{ CKA_ID, (CK_BYTE_PTR)g_id, sizeof(g_id) },
@@ -3414,7 +3414,7 @@ static void xtest_pkcs11_test_1014(ADBG_Case_t *c)
 		{ CKA_DECRYPT, &g_decrypt, sizeof(CK_BBOOL) },
 		{ CKA_WRAP, &g_wrap, sizeof(CK_BBOOL) },
 		{ CKA_UNWRAP, &g_unwrap, sizeof(CK_BBOOL) },
-		{ CKA_VALUE_LEN, &g_len, sizeof(CK_ULONG) },
+		{ CKA_VALUE_LEN, &g_len, sizeof(g_len) },
 	};
 	CK_ATTRIBUTE set_template[] = {
 		{ CKA_LABEL, (CK_UTF8CHAR_PTR)new_label, strlen(new_label) },
-- 
2.25.1

